pkg/controller/enterprisesearch/config.go

```diff
@@ -224,43 +224,51 @@ func configRefWatchName(ent types.NamespacedName) string {
 	return fmt.Sprintf("%s-%s-configref", ent.Namespace, ent.Name)
 }
 
-// parseConfigRef builds a single merged CanonicalConfig from the secrets referenced in configRef,
-// and ensures watches are correctly set on those secrets.
+// parseConfigRef builds a CanonicalConfig from the secret referenced in configRef,
+// and ensures a watch is set on this secret.
 func parseConfigRef(driver driver.Interface, ent entv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
```

Comment:
Maybe it would be worth to pull it out to common at some point?

Reply:
done in 80e410a